### ISSUE TITLE
Virtual server gets recreated due to mismatch when checking defaults

### ISSUE THREAD

**Issue by russokj** [f5devcentral/f5-cccl#198]
It appears we fail to verify the request config matches the Big-IPs config for the following request (this came from the k8s controller):
```
{
  ...
  "resources": {
    "Part-2": {
      "virtualServers": [
        {
          "name": "default_test-svc-43-map",
          "destination": "/Part-2/10.10.1.1:80",
          "source": "0.0.0.0/0",
          "enabled": true,
          "ipProtocol": "tcp"
        }
      ]
    }
  }
}
```
The following compare occurs and fails (extra print statements were added):
```
[2017-12-01 15:06:30,702 f5_cccl.service.manager DEBUG] Getting virtual address tasks...
[2017-12-01 15:06:30,702 f5_cccl.service.manager DEBUG] Getting virtual server tasks...

Virtual Server
SELF:
{'destination': '/Part-2/10.10.1.1%2:80', 'name': u'default_test-svc-43-map', 'rules': [], 'vlansDisabled': True, 'enabled': True, 'sourceAddressTranslation': {}, 'partition': u'Part-2', 'source': '0.0.0.0%2/0', 'profiles': [], 'connectionLimit': 0, 'ipProtocol': u'tcp', 'vlans': [], 'policies': []}
OTHER:
{'destination': '/Part-2/10.10.1.1%2:80', 'name': u'default_test-svc-43-map', 'rules': [], 'vlansDisabled': True, 'enabled': True, 'sourceAddressTranslation': {u'type': u'none'}, 'partition': u'Part-2', 'source': '0.0.0.0%2/0', 'profiles': [{'partition': u'Common', 'name': u'tcp', 'context': u'all'}], 'connectionLimit': 0, 'ipProtocol': u'tcp', 'vlans': [], 'policies': []}
Is Equal: False

[2017-12-01 15:06:30,702 f5_cccl.service.manager DEBUG] Getting pool tasks...
[2017-12-01 15:06:30,702 f5_cccl.service.manager DEBUG] Getting iRule tasks...
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Getting InternalDataGroup tasks...
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Getting policy tasks...
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Getting iApp tasks...
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Getting monitor tasks...
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Building task lists...
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Service task queue length: 1
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Creating 0 resources...
[2017-12-01 15:06:30,703 f5_cccl.service.manager DEBUG] Updating 1 resources...
[2017-12-01 15:06:30,703 f5_cccl.resource.resource INFO] Updating ApiVirtualServer: /Part-2/default_test-svc-43-map
[2017-12-01 15:06:30,703 root DEBUG] get WITH uri: https://10.190.24.209:443/mgmt/tm/ltm/virtual/~Part-2~default_test-svc-43-map AND suffix:  AND kwargs: {}
```

**Comment by edarzins** [comment]
The equality comparison for Virtual Servers is not robust enough. It needs to account for default values and ignore comparison of optional items that are not present in the source config.

**Comment by edarzins** [comment]
Fixed via aa4f14ae96edb74d97463d16aadd85a502d53e59.
